src/Microsoft.OData.Client/ODataRequestMessageWrapper.cs

```diff
@@ -161,6 +161,12 @@ internal static ODataRequestMessageWrapper CreateRequestMessageWrapper(BuildingR
             if (requestInfo.Timeout != 0)
             {
                 requestMessage.Timeout = requestInfo.Timeout;
+               
+            }
+
+            if (requestInfo.ReadWriteTimeout != 0)

```

Comment:
why shall we exclude the "0"?
I mean what's the effect if we directly copy the timeout ?

Reply:
so we check if the user set this value,, if they did not set it then it is zero. so if the user set the value then that value is the one that is used. Otherwise <code> HttpWebRequest </code> uses its default value.